### PR TITLE
chore(test): fix dataset test for unexpected .dataset_tmp dir in the code tree

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
image

## Modules
- [x] Client

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Review by anda-ren — DISMISSED** [review]
LGTM

File: client/tests/sdk/test_dataset.py [hunk 1]
```diff
@@ -1511,9 +1508,15 @@ def test_row(self) -> None:
 
 
 class TestRowWriter(BaseTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.workdir = Path(self.local_storage) / ".dataset_tmp"
```
Comment (jialeicui) [inline client/tests/sdk/test_dataset.py:1513]: Is it better to use `os.chdir`? 
And then there is no need to specify the `workdir` param in each test.

Comment (tianweidut) [reply client/tests/sdk/test_dataset.py]: updated

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**